homeassistant/components/stream/worker.py

```diff
@@ -37,39 +37,38 @@ def __init__(self, outputs_callback) -> None:
         self._segment_start_pts = None
         self._memory_file: BytesIO = cast(BytesIO, None)
         self._av_output: av.container.OutputContainer = None
-        self._segment: Segment = cast(Segment, None)
         self._input_video_stream: av.video.VideoStream = None
-        self._input_audio_stream = None  # type=Optional[av.audio.AudioStream]
+        self._input_audio_stream = None  # av.audio.AudioStream | None
         self._output_video_stream: av.video.VideoStream = None
-        self._output_audio_stream = None  # type=Optional[av.audio.AudioStream]
-
-    def reset_av(self) -> None:
+        self._output_audio_stream = None  # av.audio.AudioStream | None
+
+    @staticmethod
+    def make_new_av(
+        sequence: int, input_vstream: av.video.VideoStream, input_astream: Any
+    ) -> tuple[
+        BytesIO, av.container.OutputContainer, av.video.VideoStream, Any
+    ]:  # fix audio stream types later
         """Reset the av objects for this segment."""
-        self._memory_file = BytesIO()
-        self._av_output = av.open(
-            self._memory_file,
+        memory_file = BytesIO()
+        av_output = av.open(
+            memory_file,
             mode="w",
             format=SEGMENT_CONTAINER_FORMAT,
             container_options={
                 # Removed skip_sidx - see https://github.com/home-assistant/core/pull/39970
                 # "cmaf" flag replaces several of the movflags used, but too recent to use for now
                 "movflags": "frag_custom+empty_moov+default_base_moof+frag_discont+negative_cts_offsets+skip_trailer",
                 "avoid_negative_ts": "disabled",
-                "fragment_index": str(self._sequence + 1),
-                "video_track_timescale": str(
-                    int(1 / self._input_video_stream.time_base)
-                ),
+                "fragment_index": str(sequence + 1),
+                "video_track_timescale": str(int(1 / input_vstream.time_base)),
             },
         )
-        self._output_video_stream = self._av_output.add_stream(
-            template=self._input_video_stream
-        )
+        output_vstream = av_output.add_stream(template=input_vstream)
```

Comment:
I was initially expecting you'd `return av_output` then have all of these methods in  `reset`. You'd have to pass in memory file but could have fewer return arguments and not need to pass in the input astream.

Reply:
(Do you agree? I'm thinking this was my last nit.)

Reply:
Makes sense, let me make that change.
BTW, for some reason my dev docker has been really slow (takes minutes to do a lot of simple things). Have you encountered the same problem? I think it might be WSL2 related